### DISCUSSION TITLE
Calendar field type display one format, but store another

### DISCUSSION THREAD

**Discussion by dformdotdk** [joomengine/Joomla-Component-Builder#709] [Ideas]
When working with the calendar field type we normally need to store the value in the database as in this format 0000-00-00 00:00:00 (%Y-%m-%d %H:%I.%S).

However, JCB allow us to translate the format, to what is native to the user, e.g. %d-%m-%Y %H:%I.%S. But if we use translate format, then the field will not save to the database, as it will try to save in the translated format and not the format set in the format property.

In other words, I would like to be able to translate the format when displayed to the user, like e.g. 23-03-2021 22:32:37, but store it in the database as 2021-03-23 22:32:37.

I'm sure there is a way to do this, but it would be awesome if JCB could support this out of the box, when translate format was set to true.

**Comment by Llewellynvdm** [comment]
I don't understand I pointed you to a [post here in discussions](https://github.com/vdm-io/Joomla-Component-Builder/discussions/672) that answers this question, did you check it out? if yes what am I missing?

**Reply by dformdotdk** [reply]
The link you provided was not directly to that discussion, but found it my self just a few minutes ago. It does solve my issue, but it would be nice if JCB could do this when translate format was set to true.

**Reply by Llewellynvdm** [reply]
The link showed you how to search for it lol... sorry man. Me trying to be funny is not always helpful... but ok I see your post below. Here is what I think...

That is to-much "magic" --hate-the-word but... allowing you to do that in the [way I explained here](https://github.com/vdm-io/Joomla-Component-Builder/discussions/672) is the best path, and if your doing it often, you can create a customcode snippet and just bump the placeholder in here... is how I work. So I am building up my own reusable set of snippets over the years.

Reality is that there is just to many use-cases here.

**Reply by dformdotdk** [reply]
I understand. 

Noticed one thing though - when translate format is set to true, the format is changed correctly on the item, but not in the list view. Is this expected behavior or a bug?

Item:
<img width="472" alt="Skærmbillede 2021-03-23 kl  23 51 30" src="https://user-images.githubusercontent.com/6020984/112229222-dc776e00-8c32-11eb-95db-101302cde47f.png">
List:
<img width="486" alt="Skærmbillede 2021-03-23 kl  23 51 40" src="https://user-images.githubusercontent.com/6020984/112229219-db464100-8c32-11eb-9a45-eb56bdf86aa0.png">

**Reply by Llewellynvdm** [reply]
You will see in the [same post](https://github.com/vdm-io/Joomla-Component-Builder/discussions/672) that @marcorensch point to this video https://www.youtube.com/watch?v=fau5mZ6naLc&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=28 and then gives this [solution](https://github.com/vdm-io/Joomla-Component-Builder/discussions/672#discussioncomment-397209).

So it not a bug.. when you go into custom code, JCB steps away, and you need to model and handle the outcome as needed.

Meaning in this case, you need to on display of the value model the value as needed.

**Reply by Llewellynvdm** [reply]
Hmm I just looked at this video... and realize you still need this one to get the full picture:
https://www.youtube.com/watch?v=hh4IIPmYIY8&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=69

I use the relations area for all list type of manipulations... very powerful area.

**Reply by dformdotdk** [reply]
Thank you! I understand now, and have fixed my date display on the list view.

**Comment by dformdotdk** [comment]
The solution is actually given here: https://github.com/vdm-io/Joomla-Component-Builder/discussions/672 
However, it would be nice, if this was done by JCB, if translate format was set to true.

Current solution: Set modelling mode to export and add the code on save model

	if (isset([[[field]]]) && ([[[field]]] != '0000-00-00 00:00:00') && ([[[field]]] != ''))
	{
		[[[field]]] = JFactory::getDate([[[field]]])->toSql();
	}

Set on get model to: // Not needed

<img width="933" alt="Skærmbillede 2021-03-23 kl  23 39 51" src="https://user-images.githubusercontent.com/6020984/112228152-1e9fb000-8c31-11eb-897b-47cdd54b3ea5.png">
